sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/SuppressFinalizeUseless.CSharp9.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+
+GC.SuppressFinalize(new object()); // Compliant - class is static. Should we raise if `this` is not passed as parameter?
```

Comment:
> Should we raise if `this` is not passed as parameter?

this seems like a different rule? This rule is ""GC.SuppressFinalize" should not be invoked for types without destructors"

Reply:
I've noticed that [CA1816](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1816#cause) raises an issue for this but I like your suggestion too.